src/compiler/checker.ts

```diff
@@ -10891,9 +10891,9 @@ module ts {
             getSymbolDisplayBuilder().buildTypeDisplay(getReturnTypeOfSignature(signature), writer, enclosingDeclaration, flags);
         }
 
-        function isUnknownIdentifier(location: Node, name: string): boolean {
+        function isUnknownIdentifier(location: Node, name: string, sourceFile: SourceFile): boolean {

```

Comment:
I changed this so it no longer takes a source file. It now asserts that the location passed in is not synthesized.

In the emitter, I've changed emitDestructuring to take an optional parameter lowestNonSynthesizedAncestor, and I've added a comment as to how it should be used:
```ts
/**
 * If the root has a chance of being a synthesized node, callers should also pass a value for
 * lowestNonSynthesizedAncestor. This should be an ancestor of root, it should not be synthesized,
 * and there should not be a lower ancestor that introduces a scope. This node will be used as the
 * location for ensuring that temporary names are unique.
 */

```